base_local_planner/src/goal_functions.cpp

```diff
@@ -103,12 +104,8 @@ namespace base_local_planner {
     try {
       // get plan_to_global_transform from plan frame to global_frame
       tf::StampedTransform plan_to_global_transform;
-      tf.waitForTransform(global_frame, ros::Time::now(),
-                          plan_pose.header.frame_id, plan_pose.header.stamp,
-                          plan_pose.header.frame_id, ros::Duration(0.5));
-      tf.lookupTransform(global_frame, ros::Time(),
```

Comment:
I wonder if would be desirable to keep this as ros::Time() which will get the latest available transform (rather than a possibly outdated one). Although, it might also be that having all transforms done at the same time is better for consistency.

Reply:
We cannot just replace `ros::Time::now()` with `ros::Time()` in line
https://github.com/ros-planning/navigation/blob/43e9297d588276efa9868efe2ed56ac6bf08e180/base_local_planner/src/goal_functions.cpp#L106
as transform in line
https://github.com/ros-planning/navigation/blob/43e9297d588276efa9868efe2ed56ac6bf08e180/base_local_planner/src/goal_functions.cpp#L115
will use possibly later timestamp of current robot pose, issuing TF error 'extrapolation into the future', mentioned here #196